src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Tags/ZoneTag.cs

```diff
@@ -50,7 +50,7 @@ public static async ValueTask<Completion> WriteToAsync(List<FilterArgument> argu
 
                 if (zone is Shape shape)
                 {
-                    await shape.AddAsync(content, position);
+                    await shape.AddAsync(content.ToString(), position);
```

Comment:
passe as an `IHtmlContent` on purpose and it will be released when `WriteTo` is called